### DISCUSSION TITLE
osproxy mode, allow everything except ignored executables. Is it possible ?

### DISCUSSION THREAD

**Discussion by redsigma** [mitmproxy/mitmproxy#6424] [Q&A] [answered]
I know I can run `osproxy` for every application or only for a series of applications:

`mitmweb --mode osproxy:msedge.exe` (only msedge will be monitored)

`mitmweb --mode osproxy:msedge.exe,firefox.exe` (both firefox and msedge will be caught)

Is there a way to specify something like `Allow everything but ignore these processes` ? I was thinking this syntax would work:
`mitmweb --mode osproxy:-msedge.exe`  (adding a minus sign before the process name)

However it doesnt work, instead it results in a infinite page loading.

The reason why I need this is because I want to prevent browsers such as `msedge` or `firefox` to be renamed to something else like `asd.exe` thus circumventing the whole mitmproxy setup.  I cannot let the mimtproxy run for everything because I have some processes that might freak out if some app is modifying the packets (there is a chance I might misunderstand how mitmproxy works).

**Comment by mhils** [comment, marked as answer]
You can prefix with "!" to achieve that. That being said, oxproxy mode is
still highly experimental and subject to change. Feedback is welcome of
course. :)

**Reply by redsigma** [reply]
thank you so much, it worked.

So far my experience with `osproxy` was perfect. I am trying to build a browser agnostic adblocker and this tool is the best.
